src/main/scala/com/banzaicloud/spark/metrics/SparkCollectorDecorators.scala

```diff
@@ -6,15 +6,14 @@ import com.banzaicloud.spark.metrics.CollectorDecorator.FamilyBuilder
 import com.banzaicloud.spark.metrics.PushTimestampDecorator.PushTimestampProvider
 import com.codahale.metrics.MetricRegistry
 import io.prometheus.client.Collector.MetricFamilySamples
-import io.prometheus.client.Collector.MetricFamilySamples.Sample
 import io.prometheus.client.dropwizard.DropwizardExports
 import io.prometheus.jmx.JmxCollector
 
 import scala.collection.JavaConverters._
 import scala.util.matching.Regex
 
 object NameDecorator {
-  case class Replace(Regex: Regex, replacement: String)
+  case class Replace(Regex: Regex, replacement: String, toLowerCase: Boolean = false)
```

Comment:
I really wouldn't put this here: coherence is low, semantics are unintuitive (or at least ambiguous). A clearer implementation would be to create a case object `ToLowerCase` and handle it in the `NameDecorator` trait.

The best solution would be to refactor name decoration to be composable from different name decorators (replacer, case changer, etc...).